### ISSUE TITLE
`Layout/FirstArrayElementIndentation` can infinite-loop with `Layout/ArrayAlignment: with_fixed_indentation`

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15638]
## Description:

`Layout/FirstArrayElementIndentation` can fight with `Layout/ArrayAlignment` when fixed indentation is enabled.
The two cops keep changing the same array indentation back and forth, which causes an infinite autocorrect loop.

## Expected behavior:

`Layout/FirstArrayElementIndentation` should not enter an autocorrect loop with `Layout/ArrayAlignment` when `EnforcedStyle: with_fixed_indentation` is configured.

This file should autocorrect cleanly:
```ruby
foo bar: [
      1,
      2
],
baz: 3
```

with this configuration:
```yaml
AllCops:
  NewCops: disable

Layout/ArrayAlignment:
  EnforcedStyle: with_fixed_indentation
Layout/FirstArrayElementIndentation:
  EnforcedStyle: consistent
```

## Actual behavior:

`Layout/FirstArrayElementIndentation` and `Layout/ArrayAlignment` disagree about the indentation base for this case.

`Layout/FirstArrayElementIndentation` still treats the array as indentation relative to the parent hash key, while `Layout/ArrayAlignment` with `EnforcedStyle: with_fixed_indentation` corrects the same array relative to the line where the array starts.

As a result, RuboCop enters an autocorrect loop and reports:
```text
Infinite loop detected in example.rb and caused by Layout/FirstArrayElementIndentation -> Layout/ArrayAlignment

1 error occurred:
Infinite loop detected in example.rb and caused by Layout/FirstArrayElementIndentation -> Layout/ArrayAlignment
```

## Steps to reproduce the problem:

Create `.rubocop.yml`:
```yaml
AllCops:
  NewCops: disable

Layout/ArrayAlignment:
  EnforcedStyle: with_fixed_indentation
Layout/FirstArrayElementIndentation:
  EnforcedStyle: consistent
```

Create `example.rb`:
```ruby
foo bar: [
      1,
      2
],
baz: 3
```

Run:
```console
bundle exec ruby exe/rubocop example.rb --autocorrect --only Layout/ArrayAlignment,Layout/FirstArrayElementIndentation -f simple
```

Current `master` enters an infinite autocorrect loop for this case.

## RuboCop version:
```console
1.90.0 (using Parser 3.3.12.0, rubocop-ast 1.50.0, analyzing as Ruby 2.7, running on ruby 3.3.11) +Rubydex [x86_64-linux]
  - rubocop-performance 1.27.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```
